ripc-rxjava1/src/main/java/io/ripc/rx/protocol/tcp/RxConnection.java

```diff
@@ -0,0 +1,64 @@
+package io.ripc.rx.protocol.tcp;
+
+import io.ripc.protocol.tcp.Connection;
+import io.ripc.protocol.tcp.Connection.FlushSelector;
+import rx.Observable;
+import rx.RxReactiveStreams;
+import rx.Subscriber;
+
+import static rx.RxReactiveStreams.*;
+
+/**
+ * An adapter for {@link Connection} representated as an {@link Observable}
+ *
+ * @param <R> The type of objects read from this connection.
+ * @param <W> The type of objects written to this connection.
+ */
+public class RxConnection<R, W> extends Observable<R> {
+
+    private final Connection<R, W> delegate;
+
+    protected RxConnection(final Connection<R, W> delegate) {
+        super(new OnSubscribe<R>() {
+            @Override
+            public void call(Subscriber<? super R> subscriber) {
+                RxReactiveStreams.subscribe(delegate, subscriber);
+            }
+        });
+        this.delegate = delegate;
+    }
+
+    /**
+     * Writes the passed stream of {@code data} and returns the result as an {@link Observable}. All items emitted by
+     * this stream are flushed on completion of the stream.
+     *
+     * @param data Data stream to write.
+     *
+     * @return Result of write.
+     */
+    public Observable<Void> write(Observable<W> data) {
+        return toObservable(delegate.write(toPublisher(data)));
```

Comment:
> For Reactor we address this by using a Stream to represent input and output (something that's only created once) and we use a Codec facility to manage message transformation (something that's also only create once unless not practical).

Here the object allocation is because of the transformation b/w `Publisher` and `Observable` (and other associated objects) Even with a `Stream` abstraction, how will it solve the issue? We would still have to convert from `Stream` to `Observable`. 

> I was envisioning a "write" facility in this that isn't like write(byte[]) but like sink(ByteProducer). Namely an operation you could set up once and then pipe data into.

How is `sink(ByteProducer)` different than `sink(Publisher)`?

> Now that I think about (and play with) it, a Connection is simply a Processor<R,W>

Not really because `write(Publisher)` returns the result of the write where as `onNext(Producer)` does not.